environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java

```diff
@@ -116,23 +203,114 @@ public Weld(String containerId) {
      *
      * @param extension an extension
      */
-    public void addExtension(Extension extension) {
+    public Weld addExtension(Extension extension) {
         if (extensions == null) {
             extensions = new HashSet<Metadata<Extension>>();
         }
         extensions.add(new MetadataImpl<Extension>(extension, "<explicitly-added>"));
+        return this;
+    }
+
+    /**
+     *
+     * @param interceptorClasses Interceptor classes enabled for a synthetic bean archive
+     * @return self
+     */
+    public Weld enableInterceptors(Class<?>... interceptorClasses) {
+        enabledInterceptors.clear();
+        for (Class<?> interceptorClass : interceptorClasses) {
+            enabledInterceptors.add(syntheticMetadata(interceptorClass));
+        }
+        return this;
+    }
+
+    /**
+     *
+     * @param decoratorClasses Decorator classes enabled for a synthetic bean archive
+     * @return self
+     */
+    public Weld enableDecorators(Class<?>... decoratorClasses) {
+        enabledDecorators.clear();
+        for (Class<?> decoratorClass : decoratorClasses) {
+            enabledDecorators.add(syntheticMetadata(decoratorClass));
+        }
+        return this;
+    }
+
+    /**
+     *
+     * @param alternativeClasses Alternative classes selected for a synthetic bean archive
+     * @return self
+     */
+    public Weld selectAlternatives(Class<?>... alternativeClasses) {
+        selectedAlternatives.clear();
+        for (Class<?> alternativeClass : alternativeClasses) {
+            selectedAlternatives.add(syntheticMetadata(alternativeClass));
+        }
+        return this;
     }
 
     /**
-     * Boots Weld and creates and returns a WeldContainer instance, through which
-     * beans and events can be accessed.
      *
-     * @return weld container
+     * @param alternativeStereotypeClasses Alternative stereotypes selected for a synthetic bean archive
+     * @return self
+     */
+    public Weld selectAlternativeStereotypes(Class<?>... alternativeStereotypeClasses) {
```

Comment:
if BeansXml is the only concern then we can filter out annotations when building BeansXml